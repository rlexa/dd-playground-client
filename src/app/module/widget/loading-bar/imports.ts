import { CommonModule } from '@angular/common';
import { MatProgressBarModule } from '@angular/material';

export const imports = [CommonModule, MatProgressBarModule];
