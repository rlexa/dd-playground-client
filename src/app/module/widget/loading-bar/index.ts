export * from './loading-bar.module';
