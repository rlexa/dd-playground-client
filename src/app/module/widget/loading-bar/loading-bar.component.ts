import {CommonModule} from '@angular/common';
import {ChangeDetectionStrategy, Component, HostBinding, Input} from '@angular/core';
import {MatProgressBarModule} from '@angular/material/progress-bar';

@Component({
  selector: 'app-loading-bar',
  template: `@if (show) {
    <mat-progress-bar mode="indeterminate" />
  }`,
  changeDetection: ChangeDetectionStrategy.OnPush,
  standalone: true,
  imports: [CommonModule, MatProgressBarModule],
})
export class LoadingBarComponent {
  @HostBinding('style.display') readonly styleDisplay = 'block';
  @HostBinding('style.marginTop') styleMarginTop = '1rem';
  @HostBinding('style.minHeight') readonly styleMinHeight = '.5rem';

  @Input() set withMargin(val: boolean) {
    this.styleMarginTop = !!val ? '1rem' : null;
  }

  @Input() show?: boolean | null;
}
