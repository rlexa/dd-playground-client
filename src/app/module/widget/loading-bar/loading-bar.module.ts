import { NgModule } from '@angular/core';
import { imports } from './imports';
import { LoadingBarComponent } from './loading-bar.component';

@NgModule({ imports, exports: [LoadingBarComponent], declarations: [LoadingBarComponent] })
export class LoadingBarModule { }
