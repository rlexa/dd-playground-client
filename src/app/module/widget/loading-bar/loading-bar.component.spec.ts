import {ComponentFixture} from '@angular/core/testing';
import {MockBuilder, MockRender} from 'ng-mocks';
import {LoadingBarComponent} from './loading-bar.component';

describe('LoadingBarComponent', () => {
  let fixture: ComponentFixture<LoadingBarComponent>;

  beforeEach(() => MockBuilder(LoadingBarComponent));

  beforeEach(() => {
    fixture = MockRender(LoadingBarComponent);
    fixture.detectChanges();
  });

  test('renders', () => expect(fixture).toMatchSnapshot());

  describe(`with show`, () => {
    beforeEach(() => {
      fixture.componentInstance.show = true;
      fixture.detectChanges();
    });

    test('renders', () => expect(fixture).toMatchSnapshot());

    describe(`with margin`, () => {
      beforeEach(() => {
        fixture.componentInstance.withMargin = true;
        fixture.detectChanges();
      });

      test('renders', () => expect(fixture).toMatchSnapshot());
    });
  });
});
