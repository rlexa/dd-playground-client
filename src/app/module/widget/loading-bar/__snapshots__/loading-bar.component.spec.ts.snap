// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingBarComponent is created 1`] = `
<app-loading-bar
  __ngContext__={[Function LRootView]}
  show="false"
  styleDisplay={[Function String]}
  styleMarginTop={[Function String]}
  styleMinHeight={[Function String]}
>
  
</app-loading-bar>
`;
